apps/server/test/lexical/server/code_intelligence/entity_test.exs

```diff
@@ -32,8 +32,13 @@ defmodule Lexical.Server.CodeIntelligence.DefinitionTest do
   defp subject_module(project, content) do
     uri = subject_module_uri(project)
 
-    with :ok <- Document.Store.open(uri, content, 1) do
-      Document.Store.fetch(uri)
+    case Document.Store.open(uri, content, 1) do
+      :ok ->
+        Document.Store.fetch(uri)
+
+      {:error, :already_open} ->
+        :ok = Document.Store.close(uri)
+        subject_module(project, content)
```

Comment:
why not just fetch it rather than closing and recursively calling this function?
```suggestion
        Document.Store.fetch(uri)
```

Reply:
Also, this smells a little funny, why are you opening something twice in a unit test?